### DISCUSSION TITLE
TTS Response Streaming

### DISCUSSION THREAD

**Discussion by allanbunch** [allanbunch/node-red-openai-api#23] [General]
Hi everyone! I'm looking into [implementing TTS response streaming](https://discourse.nodered.org/t/https-flows-nodered-org-node-inductiv-node-red-openai-api/89052) for the node's Audio >> Create Speech method. Keep an eye on this thread for updates.

**Comment by amilanov75** [comment]
That's great news. Thank you Allan!

I played around with a direct connection, but couldn't figure out the magic combination to get my node to output anything other than a single msg with the entire file.

**Reply by allanbunch** [reply]
Curious to know if your function node direct connection attempts were similar to what's mentioned [here](https://github.com/openai/openai-node/issues/487#issuecomment-1809360671).

**Comment by amilanov75** [comment]
Hi Allan,

For some reason I cannot see your comment below on the comments page, so assume it is private to me.

No, I have not tried what is on that page. Given some time this weekend I can look into it if you would like / if it would help?

Regards
Aleks

________________________________

**Reply by allanbunch** [reply]
Sounds good, Aleks. The approach suggested may be what you're looking for. Let me know if you discover otherwise.

**Comment by amilanov75** [comment]
Thank you for the prompt reply.

I'll set aside some time (hopefully this weekend) to look into it and will report back.

________________________________

**Comment by amilanov75** [comment]
You pointed me in the exact right direction. Thank you!

I ended up getting tts to work both with a bespoke streaming text input from openai assistants API, as chunks of text, and streaming writing to the MP3 per chunk from the tts API.

It was quite a bit of work with many pitfalls and lots of design decisions on my behalf, which means there are likely other ways to get to a similar end point. 

Thanks for your help!

**Reply by allanbunch** [reply]
Great to hear! Good luck with your project!
